app/containers/sync/index.js

```diff
@@ -0,0 +1,69 @@
+import { connect } from 'react-redux';
+import { getAreas } from 'redux-modules/areas';
+import { getFeedbackQuestions } from 'redux-modules/feedback';
+import { getReportQuestions } from 'redux-modules/reports';
+import { getCountries } from 'redux-modules/countries';
+import { getLanguage } from 'helpers/language';
+import { getReadyState } from 'helpers/sync';
+import { getUser } from 'redux-modules/user';
+
+import Sync from 'components/sync';
+
+const getForms = ({ hasForms, languageChanged, getFormsDispatch }) => {
+  if (!hasForms.report || languageChanged) getFormsDispatch.report();
+  if (!hasForms.daily || languageChanged) getFormsDispatch.daily();
+  if (!hasForms.weekly || languageChanged) getFormsDispatch.weekly();
+};
+
+function mapStateToProps(state) {
+  return {
+    isConnected: state.offline.online,
+    reach: state.app.netInfo.reach,
+    languageChanged: state.app.language !== getLanguage(),
+    hasAreas: state.areas.data && state.areas.data.length > 0 && state.areas.synced,
+    hasUserData: state.user.data && Object.keys(state.user.data).length > 0 && state.user.synced,
+    hasForms: {
+      report: Object.keys(state.reports.forms).length > 0 && state.reports.synced,
```

Comment:
I think we can set the sync flag as false by default and remove the forms.length checks to make it easier to understand and avoid crashes if the forms data is not an object so I would change the name as well calling it `syncPending` or something like that.